### PR TITLE
Complete Rails Capstone Project

### PR DESCRIPTION
**Pull request by lukman155**
This PR will complete the following requirements:
- [x] Use ruby gems as software packages system.
- [x] Install Ruby on Rails framework.
- [x] Understand Rails RESTful design and router.
- [x] Use controllers to handle requests and render empty views.
- [x] Use params from browser request in a safe way.
- [x] Use preprocessed html file with embedded Ruby code.
- [x] Use layouts and templates for shared content.
- [x] Use database migration files to maintain database schema.
- [x] Use validations for models.
- [x] Secure app from n+1 problems.
- [x] Understand what ORM is.
- [x] Write SQL queries with ActiveRecord.
- [x] Set up associations between models.
- [x] Build a webapp that requires the user to log in.
- [x] Use devise gem for authentication.
- [x] Limit access to webapp resources based on authorization rules.
- [x] Analyze in writing why you have made a coding choice using one structure over another.

### CODE REVIEW

File: app/views/splash/show.html.erb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  <title>Splash Screen</title>
+  <%= stylesheet_link_tag 'splash' %>
+</head>
+<body>
+  <div class="buttons">
+    <%= link_to "Log in", new_user_session_path, class: "button" %>
+    <%= link_to "Sign up", new_user_registration_path, class: "button" %>
+  </div>
+</body>
+</html>
```
Comment (zieeco) [inline app/views/splash/show.html.erb]: - Please provide the name of your app on splash page and it should be in the center of the page as it is designed in the app.

*main design*
image

*your design*
image

File: app/views/categories/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+        <h2 class="card-title"><%= category.name %></h2>
+        <p class="card-text">Total Amount: <%= category.bills.sum(:amount) %></p>
+      </div>
+    <% end %>
+  <% end %>
+</div>
+
+<div class="d-flex justify-content-end">
+  <%= link_to "New category", new_category_path, class: "new-bill-button"  %>
+</div>
```
Comment (zieeco) [inline app/views/categories/index.html.erb]: - Hey, let's adjust our styling here to match the alignment and placement of the contents on each card item. see the image below:
- Add the missing dollar sign.
- Remove the extra added text `Total amount`

*your design*
image

*main design*
image

File: spec/models/user_spec.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  describe 'associations' do
+    it { should have_many(:bills).with_foreign_key('author_id') }
+    it { should have_many(:categories) }
+  end
+end
```
Comment (zieeco) [inline spec/models/user_spec.rb]: - Hey, let's add a test that shows the user has a name as you've written in the user model.

File: spec/models/category_spec.rb [hunk 4]
```diff
@@ -0,0 +1,8 @@
+require 'rails_helper'
+
+RSpec.describe Category, type: :model do
+  describe 'associations' do
+    it { should belong_to(:user) }
+    it { should have_many(:bills) }
+  end
```
Comment (zieeco) [inline spec/models/category_spec.rb:7]: - Hey, let's also add tests that show the category has an icon and a name as in the category model.

image

File: spec/models/bill_spec.rb [hunk 5]
```diff
@@ -0,0 +1,7 @@
+require 'rails_helper'
+
+RSpec.describe Bill, type: :model do
+  describe 'associations' do
+    it { should belong_to(:author).class_name('User') }
+  end
```
Comment (zieeco) [inline spec/models/bill_spec.rb:6]: - Hey, let's also add tests that show the bill has a name and an amount as in the bill model.


image

**Review by zieeco — CHANGES_REQUESTED** [review]
Hi @lukman155,





Great job so far! 👏 
You've done well, however, there are some issues that you still need to work on to go to the next project, but you are almost there!


## To highlight 🎯

✅ Descriptive PR 👍 
✅ Nice project implementation 🔥 
✅ Good commit messages 👍 
✅ Linters are passing 👏 

## Required Changes ♻️
_Check the comments under the review._




## Optional suggestions
- N/A



**_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were missed by the previous reviewer and addressing them will really improve your application._**


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by codecaiine — APPROVED** [review]
Hi @lukman155 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- good job implementing the requested changes ✔️
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me (@codecaiine) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**